pkg/rngd/cmd/rngd/rng_linux_arm64.go

```diff
@@ -7,15 +7,47 @@ package main
 import "C"
 
 import (
-	"errors"
+	"encoding/binary"
+	"log"
+	"os"
 )
 
-// No standard RNG on arm64
-
+// Check if there is a hardware RNG device on AArch64, will use that
+// device as the RNG entropy source if we do have
 func initRand() bool {
-	return false
+
+	f, err := os.Open("/dev/hwrng")
+	defer f.Close()
+	if err != nil {
+		log.Fatalf("Cannot open /dev/hwrng: %v", err)
+		return false
+	}
+
+	// Try to read some data from the entropy source. If it doesn't
+	// return an error, assume it's ok to use
+	buf := make([]byte, 16)
+	n, err := f.Read(buf)
+	if err != nil || n != 16 {
+		log.Fatalf("Read /dev/hwrng failed: %v", err)
+		return false
+	}
+
+	//	littleEndian = isLittleEndian()
+	return true
 }
 
 func rand() (uint64, error) {
-	return 0, errors.New("No randomness available")
+	var r uint64
+	f, _ := os.Open("/dev/hwrng")
+	defer f.Close()
+
+	buf := make([]byte, 8)
+	_, err := f.Read(buf)
+	if err != nil {
+		return 0, err
+	}
+
+	r = binary.LittleEndian.Uint64(buf)
```

Comment:
OOI why is the endianess of the random data relevant?

Reply:
Right. Will consolidate that with Justin's comments followed.